crates/sui-json-rpc/src/balance_changes.rs

```diff
@@ -109,26 +109,25 @@ async fn fetch_coins<P: ObjectProvider<Error = E>, E>(
     let mut all_mutated_coins = vec![];
     for (id, version, digest_opt) in objects {
         // TODO: use multi get object
-        if let Ok(o) = object_provider.get_object(id, version).await {
```

Comment:
I think the original code was written like this to handle dry_RunTransactions on packages, so we need to handle this, or take balances changes out of dry_Run